Lib/idlelib/editor.py

```diff
@@ -1052,12 +1054,19 @@ def saved_change_hook(self):
         else:
             title = "untitled"
         icon = short or long or title
-        if not self.get_saved():
+        if not self.get_saved() and not macosx.isCocoaTk():
             title = "*%s*" % title
             icon = "*%s" % icon
         self.top.wm_title(title)
         self.top.wm_iconname(icon)
 
+        if macosx.isCocoaTk():
+            # Add a proxy icon to the window title
+            self.top.wm_attributes("-titlepath", long)
+
+            # Maintain the modification status for the window
+            self.top.wm_attributes("-modified", not self.get_saved())
```

Comment:
The two Apple-supplied editors on my macbook do not do this.

Reply:
Which ones? 

The change indicator is not used for applications that use Cocoa's document model and autosaving.

With Text Edit the change indicator in the window controls is never used, but change is noticed in the windows title. That's part of Cocoa's NSDocument model and cannot be reproduced with Tkinter (at least not easily).

Nova also doesn't use the indicator in the window controls, but that's because they are using a tabbed interface with a change indicator in the tab.

MS Word does use a change indicator in window title (for documents stored locally, cloud stored files have autosave enabled).